Library/Homebrew/os/mac/version.rb

```diff
@@ -20,35 +23,44 @@ class Version < ::Version
         yosemite:    "10.10",
       }.freeze
 
+      sig { params(sym: Symbol).returns(T.attached_class) }
       def self.from_symbol(sym)
         str = SYMBOLS.fetch(sym) { raise MacOSVersionError, sym }
         new(str)
       end
 
+      sig { params(value: T.nilable(String)).void }
       def initialize(value)
-        super(value)
+        raise MacOSVersionError, value unless value&.match?(/\A1\d+(?:\.\d+){0,2}\Z/)
```

Comment:
```suggestion

        raise MacOSVersionError, value unless /\A1\d+(?:\.\d+){0,2}\Z/.match?(value)
```

which will work fine with a `nil` value.